labelbox/data/annotation_types/data/tiled_image.py

```diff
@@ -0,0 +1,364 @@
+
+        # Make sure that the tiles are increasing in order
+        xstart, xend = min(xstart, xend), max(xstart, xend)
+        ystart, yend = min(ystart, yend), max(ystart, yend)
+        return (*[pt * 2.0**zoom for pt in [xstart, ystart, xend, yend]],)
+
+    def _latlng_to_tile(self,
+                        lat: float,
+                        lng: float,
+                        zoom=0) -> Tuple[float, float]:
+        """Converts lat/lng to 3857 tile coordinates
+        Formula found here:
+        https://wiki.openstreetmap.org/wiki/Slippy_map_tilenames#lon.2Flat_to_tile_numbers_2
+        """
+        scale = 2**zoom
+        lat_rad = math.radians(lat)
+        x = (lng + 180.0) / 360.0 * scale
+        y = (1.0 - math.asinh(math.tan(lat_rad)) / math.pi) / 2.0 * scale
+        return x, y
+
+    def _tile_to_pixel(self, tile: float) -> Tuple[int, int]:
+        """Rounds a tile coordinate and reports the remainder in pixels
+        """
+        rounded_tile = int(tile)
+        remainder = tile - rounded_tile
+        pixel_offset = int(self.tile_size * remainder)
+        return rounded_tile, pixel_offset
+
+    def _fetch_image_for_bounds(
+        self,
+        x_tile_start: int,
+        y_tile_start: int,
+        x_tile_end: int,
+        y_tile_end: int,
+        zoom: int,
+    ) -> np.ndarray:
+        """Fetches the tiles and combines them into a single image
+        """
+        tiles = {}
+        with ThreadPoolExecutor(max_workers=TILE_DOWNLOAD_CONCURRENCY) as exc:
+            for x in range(x_tile_start, x_tile_end + 1):
+                for y in range(y_tile_start, y_tile_end + 1):
+                    tiles[(x, y)] = exc.submit(self._fetch_tile, x, y, zoom)
+
+        rows = []
+        for y in range(y_tile_start, y_tile_end + 1):
+            rows.append(
+                np.hstack([
+                    tiles[(x, y)].result()
+                    for x in range(x_tile_start, x_tile_end + 1)
+                ]))
+
+        return np.vstack(rows)
+
+    @retry(delay=1, tries=6, backoff=2, max_delay=16)
+    def _fetch_tile(self, x: int, y: int, z: int) -> np.ndarray:
+        """
+        Fetches the image and returns an np array. If the image cannot be fetched, 
+        a padding of expected tile size is instead added.
+        """
+        try:
+            data = requests.get(self.tile_layer.url.format(x=x, y=y, z=z))
+            data.raise_for_status()
+            decoded = tf.image.decode_image(data.content, channels=3).numpy()
+            if decoded.shape[:2] != (self.tile_size, self.tile_size):
+                logger.warning(
+                    f"Unexpected tile size {decoded.shape}. Results aren't guarenteed to be correct."
+                )
+        except:
+            logger.warning(
+                f"Unable to successfully find tile. for z,x,y: {z},{x},{y} "
+                "Padding is being added as a result.")
+            decoded = np.zeros(shape=(self.tile_size, self.tile_size, 3),
```

Comment:
This will prevent the retry of the request from occurring since there will never be an exception. If you want this functionality, it needs to go where we call `.result()`

Reply:
Could you explain why this needs to be where the `.result()` is sitting? I understand it will prevent the retry of the request but think it's appropriate to be under the `_fetch_tile` method

Reply:
Could you explain why this needs to be where the `.result()` is sitting? I understand it will prevent the retry of the request but think it's appropriate to be under the `_fetch_tile` method

Reply:
Hmm, if it is included in where we call `.result()` would it be a try/except on the result, and if there is no exception, then add the padding there?